omx/gstomxcamera.c

```diff
@@ -268,6 +285,15 @@ gst_omx_camera_init (GstOMXCamera * self)
   self->always_copy = PROP_ALWAYS_COPY_DEFAULT;
   self->num_buffers = PROP_NUM_OUT_BUFFERS_DEFAULT;
   self->skip_frames = PROP_SKIP_FRAMES_DEFAULT;
+  self->provide_clock = PROP_PROVIDE_CLOCK_DEFAULT;
+  if (PROP_PROVIDE_CLOCK_DEFAULT)
+    GST_OBJECT_FLAG_SET (self, GST_ELEMENT_FLAG_PROVIDE_CLOCK);
+  else
+    GST_OBJECT_FLAG_UNSET (self, GST_ELEMENT_FLAG_PROVIDE_CLOCK);
+
+  self->clock =
+      g_object_new (GST_TYPE_OMX_CLOCK, "name", "GstOmxCameraClock", NULL);
+  g_object_set (self->clock, "clock-type", GST_CLOCK_TYPE_OTHER, NULL);
```

Comment:
This should be in the Clock class not here

Reply:
Ok, moved to gst_omx_clock_init